### DISCUSSION TITLE
FontSubstitutionRequired `Courier' @ warning/type.c/GetTypeInfoByFamily/490

### DISCUSSION THREAD

**Discussion by lalitap25** [ImageMagick/ImageMagick#7414] [Help]
### ImageMagick version

7.1.1-21

### Operating system, version and so on

linux

### Description

Hi There,
I get below error message while converting .wmf file to png file
Error output of executed command parameters [[/opt/ole-convert/oleconvert.sh, /tmp/picture9047774478101023383.wmf, /tmp/convertedPic9749647479699163009.png]]

I do have the font 
/app/ImageMagick-7.1.1-21 # identify -list font | grep -i courier
  Font: Courier
    family: Courier
  Font: Courier-Bold
    family: Courier
  Font: Courier-BoldOblique
    family: Courier
  Font: Courier-Oblique
    family: Courier
---------------
 Font: Courier
    family: Courier
    style: Normal
    stretch: Normal
    weight: 400
    glyphs: /usr/share/ghostscript/fonts/n022003l.pfb
  Font: Courier-Bold
    family: Courier
    style: Normal
    stretch: Normal
    weight: 700
    glyphs: /usr/share/ghostscript/fonts/n022004l.pfb
  Font: Courier-BoldOblique
    family: Courier
    style: Oblique
    stretch: Normal
    weight: 700
    glyphs: /usr/share/ghostscript/fonts/n022024l.pfb
  Font: Courier-Oblique
    family: Courier
    style: Oblique
    stretch: Normal
    weight: 400
    glyphs: /usr/share/ghostscript/fonts/n022023l.pfb

As a result of this issue , I get distorted png images which are really not visible 

what am I missing?

**Comment by fmw42** [comment]
Post a link to your wmf file!

**Reply by lalitap25** [reply]
[Downloads.zip](https://github.com/user-attachments/files/15908286/Downloads.zip)
this zip has 2 wmf files for sample.

**Comment by dlemstra** [comment]
We probably need to improve that warning. A font cannot be found and that is replaced with `Courier`.

**Comment by betayyang** [comment]
I have simalar questions,the error message is:
convert: FontSubstitutionRequired `Arial' @ warning/type.c/GetTypeInfoByFamily/490.

fc-list | grep Arial command shows I have installed the font

**Reply by dlemstra** [reply]
It's a warning and not an error. This warning is reporting that Arial was used as the font to substitute another font that was not found on your system.

**Reply by betayyang** [reply]
can I know what kind of font I lost?

**Reply by dlemstra** [reply]
The error message now doesn't display that we should probably change that.

**Reply by betayyang** [reply]
Is there any font install list map for imagemagick?
